cmd/juju/commands/ssh.go

```diff
@@ -102,24 +112,48 @@ func (c *sshCommand) Info() *cmd.Info {
 	})
 }
 
-func (c *sshCommand) Init(args []string) error {
+func (c *sshCommand) Init(args []string) (err error) {
 	if len(args) == 0 {
 		return errors.Errorf("no target name specified")
 	}
-	c.Target, c.Args = args[0], args[1:]
+	c.modelType, err = c.ModelType()
+	if err != nil {
+		return err
+	}
+	if c.modelType == model.CAAS {
+		c.provider = &c.sshContainer
+	} else {
+		c.provider = &c.SSHCommon
+	}
+	c.provider.SetTarget(args[0])
+	c.provider.SetArgs(args[1:])
+	c.provider.setHostChecker(c.hostChecker)
 	return nil
 }
 
+type sshProvider interface {
+	initRun(modelcmd.ModelCommandBase) error
+	cleanupRun()
+	setHostChecker(checker jujussh.ReachableChecker)
+	resolveTarget(string) (*resolvedTarget, error)
+	ssh(ctx Context, enablePty bool, target *resolvedTarget) error
+
+	GetTarget() string
+	SetTarget(target string)
+
+	GetArgs() []string
+	SetArgs(Args []string)
```

Comment:
Do these need to be exported? Everything that uses them is in the same package

Reply:
right, lower the case.